test/osb_test/osb_test.go

```diff
@@ -148,13 +154,68 @@ var _ = Describe("Service Manager OSB API", func() {
 		failingBrokerID      string
 		smUrlToFailingBroker string
 
-		smUrlToQueryVerificationBroker string
-		headerKey                      string
-		headerValue                    string
+		queryParameterVerificationServer   *common.BrokerServer
+		queryParameterVerificationServerID string
+		smUrlToQueryVerificationBroker     string
+		headerKey                          string
+		headerValue                        string
+
+		timeoutBrokerServer     *common.BrokerServer
+		timeoutBrokerID         string
+		smUrlToTimeoutBroker    string
+		timeoutBrokerReleaseCtx context.Context
+		timeoutBrokerRelease    context.CancelFunc
 	)
 
+	resetBrokersHandlers := func() {
+		validBrokerServer.ResetHandlers()
+		brokerServerWithEmptyCatalog.ResetHandlers()
+		stoppedBrokerServer.ResetHandlers()
+		failingBrokerServer.ResetHandlers()
+		queryParameterVerificationServer.ResetHandlers()
+		timeoutBrokerServer.ResetHandlers()
+
+		failingBrokerServer.ServiceInstanceHandler = failingHandler
+		failingBrokerServer.BindingHandler = failingHandler
+		failingBrokerServer.CatalogHandler = failingHandler
+		failingBrokerServer.ServiceInstanceLastOpHandler = failingHandler
+		failingBrokerServer.BindingLastOpHandler = failingHandler
+		failingBrokerServer.BindingAdaptCredentialsHandler = failingHandler
+
+		queryParameterVerificationServer.ServiceInstanceHandler = queryParameterVerificationHandler(headerKey, headerValue)
+		queryParameterVerificationServer.BindingHandler = queryParameterVerificationHandler(headerKey, headerValue)
+		queryParameterVerificationServer.CatalogHandler = queryParameterVerificationHandler(headerKey, headerValue)
+		queryParameterVerificationServer.ServiceInstanceLastOpHandler = queryParameterVerificationHandler(headerKey, headerValue)
+		queryParameterVerificationServer.BindingLastOpHandler = queryParameterVerificationHandler(headerKey, headerValue)
+
+		timeoutBrokerReleaseCtx, timeoutBrokerRelease = context.WithCancel(context.Background())
+		stoppingHandler := func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
Would it make sense to extract as constant with meaningful name instead of the comment? Something like additionalDelayAfterTimeout